pf/tfbridge/provider_read.go

```diff
@@ -90,12 +91,36 @@ func (p *provider) ReadWithContext(
 		}
 
 		// __defaults is not needed for Plugin Framework bridged providers
-		delete(result.Inputs, "__defaults")
+		deleteDefaultsKey(&result.Inputs)
 	}
 
 	return result, ignoredStatus, err
 }
 
+// deleteDefaultsKey removes the `__defaults: []` entry from all objects recursively
+// The `__defaults` key is something used in sdkv2 and is not handled here in pf. Because
+// of some code reuse between sdkv2 & pf the `__defaults` key is getting inserted
+func deleteDefaultsKey(inputs *resource.PropertyMap) {
+	i := *inputs
+	delete(i, "__defaults")
+	for _, key := range i.StableKeys() {
```

Comment:
This is good. NIt: the for loop and top level delete may have been avoided by `propertyvalue.TransformPropertyValue` on `resource.NewObjectProperty(inputs)`. But this works too.

Reply:
It looks like `TransformPropertyValue` won't  run the `transform` function on the top level value, only on each value in the map downward.